packages/gatsby-source-contentful/src/extend-node-type.js

```diff
@@ -458,6 +460,30 @@ const fluidNodeType = ({ name, getTracedSVG }) => {
 }
 
 exports.extendNodeType = ({ type, store }) => {
+  if (type.name.match(/contentful.*RichTextNode/)) {
+    return {
+      data: {
+        type: GraphQLJSON,
+        deprecationReason: `This field is deprecated, please use 'json' instead.`,
+      },
+      nodeType: {
+        type: GraphQLString,
+        deprecationReason: `This field is deprecated, please use 'json' instead.`,
+      },
+      content: {
```

Comment:
@sbezludny deprecating the content field broke existing queries mainly because `type: new GraphQLList(GraphQLJSON)` I think.
check @pieh message

Reply:
It's a shame that we can't "just" add `deprecationReason` (right now) and keep inferred type. So I think we need to drop deprecation notice here

Reply:
thanks for the suggestion @pieh I'll update the PR.